### PR TITLE
fix: update server-cluster-role.yaml

### PR DESCRIPTION
**Pull request by kristofarkas**
Checklist:

* [x] I have update the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have followed the testing instructions in the [contributing guide](https://github.com/argoproj/argo-helm/blob/master/CONTRIBUTING.md).
* [x] I have signed the CLA and the build is green.
* [x] I will test my changes again once merged to master and published.

Changes are automatically published when merged to `master`. They are not published on branches.

### CODE REVIEW

**Comment by kristofarkas** [comment]
Fixes https://github.com/argoproj/argo-helm/issues/277

File: charts/argo/templates/server-cluster-role.yaml [hunk 1]
```diff
@@ -28,6 +28,9 @@ rules:
   - get
   - list
   - watch
+  - update
+  - patch
+  - delete
```
Comment (stefansedich) [inline charts/argo/templates/server-cluster-role.yaml:33]: Should we only be adding `delete` here? looking at the manifests over in the Argo repo https://github.com/argoproj/argo/blob/3f8be0cd48963958c493e7669a1d03bb719b375a/manifests/namespace-install/argo-server-rbac/argo-server-role.yaml#L30 looks like it only has `get,list,watch,delete`.

**Comment by stefansedich** [comment]
@kristofarkas mind rebasing, resolving the conflicts and bumping the chart version.

File: charts/argo/templates/server-cluster-role.yaml (hunk 1 above)
Comment (stefansedich) [reply charts/argo/templates/server-cluster-role.yaml:33]: If you could also rebase, fix the conflicts and bump the chart version in `Chart.yaml` that would be great :smile:

Comment (kristofarkas) [reply charts/argo/templates/server-cluster-role.yaml:31]: Done!

File: charts/argo/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 appVersion: "v2.7.6"
 description: A Helm chart for Argo Workflows
 name: argo
-version: 0.8.3
+version: 0.8.5
```
Comment (stefansedich) [inline charts/argo/Chart.yaml:5]: Looks like we skipped 0.8.4, want to bump down and I will get this in!

Comment (stefansedich) [reply charts/argo/Chart.yaml]: Nevermind I was able take care of it!

**Pull request merged**